### PR TITLE
Fixed mapping conflict issues & cleaned up Elasticsearch templates.

### PR DESCRIPTION
**Pull request by kishorebanala**

### CODE REVIEW

File: es5-persistence/src/main/resources/template_tasklog.json [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+                        }
+                    }
+                },
+                "status": {
+                    "type": "string",
+                    "index": "not_analyzed"
+                }
+            }
+        },
+        "task": {
```
Comment (kishorebanala) [inline es5-persistence/src/main/resources/template_tasklog.json]: Change to task_log.

File: es2-persistence/src/main/java/com/netflix/conductor/dao/es/index/ElasticSearchDAO.java [hunk 2]
```diff
@@ -182,15 +182,15 @@ private void updateIndexName(Configuration config) {
 	 */
 	private void initIndex() throws Exception {
 
-		//0. Add the index template
-		GetIndexTemplatesResponse result = elasticSearchClient.admin().indices().prepareGetTemplates("wfe_template").execute().actionGet();
+		//0. Add the tasklog template
+		GetIndexTemplatesResponse result = elasticSearchClient.admin().indices().prepareGetTemplates("tasklog_template").execute().actionGet();
 		if (result.getIndexTemplates().isEmpty()) {
-			logger.info("Creating the index template 'wfe_template'");
```
Comment (cyzhao) [inline es2-persistence/src/main/java/com/netflix/conductor/dao/es/index/ElasticSearchDAO.java]: Just to confirm, is this no longer needed?

File: es2-persistence/src/main/java/com/netflix/conductor/dao/es/index/ElasticSearchDAO.java [hunk 3]
```diff
@@ -182,15 +182,15 @@ private void updateIndexName(Configuration config) {
...
-			InputStream stream = ElasticSearchDAO.class.getResourceAsStream("/template.json");
+			logger.info("Creating the index template 'default_template'");
```
Comment (cyzhao) [inline es2-persistence/src/main/java/com/netflix/conductor/dao/es/index/ElasticSearchDAO.java]: did you mean 'tasklog_template' here?

File: es2-persistence/src/main/java/com/netflix/conductor/dao/es/index/ElasticSearchDAO.java [hunk 4]
```diff
@@ -206,11 +206,11 @@ private void initIndex() throws Exception {
...
 		}
 
-		//2. Mapping for the workflow document type
-		GetMappingsResponse response = elasticSearchClient.admin().indices().prepareGetMappings(indexName).addTypes(WORKFLOW_DOC_TYPE).execute().actionGet();
-		if (response.mappings().isEmpty()) {
+		//2. Add Mappings for the workflow document type
+		GetMappingsResponse getMappingsResponse = elasticSearchClient.admin().indices().prepareGetMappings(indexName).addTypes(WORKFLOW_DOC_TYPE).execute().actionGet();
+		if (getMappingsResponse.mappings().isEmpty()) {
 			logger.info("Adding the workflow type mappings");
-			InputStream stream = ElasticSearchDAO.class.getResourceAsStream("/wfe_type.json");
```
Comment (cyzhao) [inline es2-persistence/src/main/java/com/netflix/conductor/dao/es/index/ElasticSearchDAO.java]: same here, not needed anymore? remove the file if yes.

File: es5-persistence/src/main/resources/template_tasklog.json [hunk 5]
```diff
@@ -0,0 +1,85 @@
+{
```
Comment (cyzhao) [inline es5-persistence/src/main/resources/template_tasklog.json]: there is only one template for *task*log*, but not for others?

File: es2-persistence/src/main/java/com/netflix/conductor/dao/es/index/ElasticSearchDAO.java (hunk 2 above)
Comment (kishorebanala) [reply es2-persistence/src/main/java/com/netflix/conductor/dao/es/index/ElasticSearchDAO.java]: Yes, tasklog_template would suffice the templates required for conductor. Others are defined in docType mappings.

File: es2-persistence/src/main/java/com/netflix/conductor/dao/es/index/ElasticSearchDAO.java (hunk 4 above)
Comment (kishorebanala) [reply es2-persistence/src/main/java/com/netflix/conductor/dao/es/index/ElasticSearchDAO.java]: Yes, files are removed.

**Pull request merged**